src/midi.c

```diff
@@ -0,0 +1,979 @@
+// Commander X16 Emulator
+// Copyright (c) 2024 MooingLemur
+// All rights reserved. License: 2-clause BSD
+
+#include <pthread.h>
+#include "glue.h"
+#include "midi.h"
+#include "audio.h"
+
+#ifdef _WIN32
+    #include <windows.h>
+    #define LIBRARY_TYPE HMODULE
+    #define LOAD_LIBRARY(name) LoadLibrary(name)
+    #define GET_FUNCTION(lib, name) GetProcAddress(lib, name)
+    #define CLOSE_LIBRARY(lib) FreeLibrary(lib)
+#else
+    #include <dlfcn.h>
+    #define LIBRARY_TYPE void*
+    #define LOAD_LIBRARY(name) dlopen(name, RTLD_LAZY)
+    #define GET_FUNCTION(lib, name) dlsym(lib, name)
+    #define CLOSE_LIBRARY(lib) dlclose(lib)
+#endif
+
+#define ASSIGN_FUNCTION(lib, var, name) {\
+    var = (void *)GET_FUNCTION(lib, name);\
+    if (!var) { fprintf(stderr, "Unable to find symbol for '%s'\n", name); CLOSE_LIBRARY(handle); return; }\
+}
+
+enum MIDI_states {
+    NORMAL,
+    PARAM,
+    SYSEX,
```

Comment:
Really?  This is not what I've found in the last couple projects I've looked at.  FluidSynth source itself uses all caps for members. as does Furnace.  Enum types themselves are usually CamelCased in Furnace and not at all in FluidSynth.

Reply:
C unfortunately puts enum members into a global namespace, so some projects use enums as "typed constants" - though the majority of I've seen use something like 
```c
enum FooBar {
    FB_Value1,
    FB_Value2
};
```

, aka CamelCase with a prefix.

Reply:
I get the impression that there is no consensus for this whatsoever.  I asked a friend which he prefers, and he says C gets all caps, and C++ gets PascalCase.

I will at least prefix them to avoid clashes in the global namespace.